packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/cloudwatch_logger_plugin.dart

```diff
@@ -139,32 +141,91 @@ class CloudWatchLoggerPlugin extends AWSLoggerPlugin
-          // log store is full.
-          break;
+        _TooNewLogEventException? tooNewException;
+        while (logs.isNotEmpty && events.isNotEmpty) {
+          final rejectedLogEventsInfo =
+              (await _sendToCloudWatch(events)).rejectedLogEventsInfo;
+          if (rejectedLogEventsInfo == null) {
+            await _logStore.deleteItems(logs);
+            break;
+          }
+          {
+            final (tooOldEndIndex, tooNewStartIndex) =
+                rejectedLogEventsInfo.parse(events.length);
+
+            if (_isValidIndex(tooNewStartIndex, events.length)) {
+              tooNewException = _TooNewLogEventException(
+                events[tooNewStartIndex!].timestamp.toInt(),
+              );
+              // set logs to end before the index.
+              logs.removeRange(tooNewStartIndex, events.length);
+              // set events to end before the index.
+              events.removeRange(tooNewStartIndex, events.length);
+            }
+            if (_isValidIndex(tooOldEndIndex, events.length)) {
+              // remove old logs from log store.
+              await _logStore.deleteItems(logs.sublist(0, tooOldEndIndex! + 1));
+              // set logs to start after the index.
+              logs.removeRange(0, tooOldEndIndex + 1);
+              // set events to start after the index.
+              events.removeRange(0, tooOldEndIndex + 1);
+            }
+            continue;
+          }
+        }
+        // after sending each batch to CloudWatch check if the batch has
+        // `tooNewException` and throw to stop syncing next batches.
+        if (tooNewException != null) {
+          throw tooNewException;
         }
-        await _logStore.deleteItems(logs);
       }
     }
 
     if (!_syncing) {
-      // TODO(nikahsn): disable log rotation.
       _syncing = true;
+      DateTime? nextRetry;
       try {
         await startSyncing();
+      } on _TooNewLogEventException catch (e) {
+        nextRetry =
+            DateTime.fromMillisecondsSinceEpoch(e.timeInMillisecondsSinceEpoch)
+                .add(_minusMaxLogEventTimeInFuture);
       } on Exception catch (e) {
         logger.error('Failed to sync logs to CloudWatch.', e);
-        // TODO(nikahsn): enable log rotation if the log store is full
       } finally {
+        _handleFullLogStoreAfterSync(
+          retryTime: nextRetry,
+        );
         _syncing = false;
       }
     }
   }
 
+  void _handleFullLogStoreAfterSync({
+    DateTime? retryTime,
+  }) {
+    final isLogStoreFull =
+        _logStore.isFull(_pluginConfig.localStoreMaxSizeInMB);
+    if (!isLogStoreFull) {
+      _retryCount = 0;
+      _retryTime = null;
+      return;
+    }
+    if (retryTime != null && retryTime.isAfter(DateTime.timestamp())) {
+      _retryTime = retryTime;
+      return;
+    }
+    _retryCount += 1;
+    _retryTime = DateTime.timestamp().add((_baseRetryInterval * _retryCount));
```

Comment:
This seems like a long time for a basic backoff

Reply:
`flushIntervalInSeconds = const Duration(seconds: 60)`. updated the `_baseRetryInterval` to 10 secs so there will be 2-3 retries within the flush interval.